src/k8s/pkg/k8sd/api/cluster_join_cleanup.go

```diff
@@ -0,0 +1,32 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microcluster/state"
+)
+
+func (e *Endpoints) deleteClusterJoin(s *state.State, r *http.Request) response.Response {
+	snap := e.provider.Snap()
+
+	req := apiv1.ClusterJoinCleanupRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		return response.BadRequest(fmt.Errorf("failed to parse request: %w", err))
+	}
+
+	// Stop k8sdqlite service and remove directory
+	snaputil.StopK8sDqliteServices(s.Context, snap)
+	os.RemoveAll(snap.K8sDqliteStateDir())
```

Comment:
please make sure that this is enough, and after removing a control plane node there is no leftover nodes in k8s-dqlite